### PR TITLE
Delete queries for orphan functional_annotation_agg records and id non-conformaing Data Objects

### PR DESCRIPTION
**Pull request by mbthornton-lbl**
Provides .js queries to find 
1) legacy data_object_set records
2) functional_annotation_agg records with a annotation activity that no longer exists

Also includes a python script to write data files for a queries:run deletion request.

Python code has been tested on runtime dev, results will be slightly different as the data is not synchronized but the requests validate.

### CODE REVIEW

File: nmdc_schema/create_legacy_orphan_deletion_requests.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import json
+import csv
+
+
+def gen_delete(delete_list: list, key: str,limit: int,del_coll: str):
```
Comment (turbomam) [inline nmdc_schema/create_legacy_orphan_deletion_requests.py:5]: adding docstrings preferable

**Review by turbomam — COMMENTED** [review]
formatting with black preferable

File: nmdc_schema/create_legacy_orphan_deletion_requests.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import json
+import csv
+
+
+def gen_delete(delete_list: list, key: str, limit: int, del_coll: str):
```
Comment (eecavanna) [inline nmdc_schema/create_legacy_orphan_deletion_requests.py:5]:
```suggestion
def gen_delete(delete_list: list, key: str, limit: int, del_coll: str):
    r"""
    Generates HTTP request bodies that can be sent to the Runtime API's
    `/queries:run` endpoint in order to delete documents from the database.
    """
```

File: nmdc_schema/create_legacy_orphan_deletion_requests.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+with open("to_delete_nmdc.functional_annotation_agg.csv", newline="") as csvfile:
+    agg_records = csv.reader(csvfile, delimiter=",")
+    for row in agg_records:
+        delete_agg_list.append(row[0])
+del_coll = "functional_annotation_agg"
+gen_delete(delete_agg_list, "metagenome_annotation_id", 0, del_coll)
+
+# delete leagacy orphan data_object_set records
+delete_do_list = []
+with open("legacy_orphan.nmdc.data_object_set.csv", newline="") as csvfile:
```
Comment (eecavanna) [inline nmdc_schema/create_legacy_orphan_deletion_requests.py:28]: Long term, I'd recommend this be a CLI argument.

Short term, I'd recommend it be a variable defined at the top of the file.

Immediate term, I'm OK with this being merged in as is.

File: nmdc_schema/create_legacy_orphan_deletion_requests.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+        delete_statement.append({"q": {key: record}, "limit": limit})
+    request_body_json = {"delete": del_coll, "deletes": delete_statement}
+    request_body_file = del_coll + "_request_body.json"
+    with open(request_body_file, "w") as f:
+        json.dump(request_body_json, f)
+
+
+# delete legacy orphan functional_annotation_agg records
+delete_agg_list = []
+with open("to_delete_nmdc.functional_annotation_agg.csv", newline="") as csvfile:
```
Comment (eecavanna) [inline nmdc_schema/create_legacy_orphan_deletion_requests.py:19]: Long term, I'd recommend this file path be a CLI argument.

Short term, I'd recommend it be a variable defined at the top of the file.

Immediate term, I'm OK with this being merged in as is.

**Review by eecavanna — APPROVED** [review]
I'm comfortable with this being merged into the `main` branch as is. I left comments about some aspects of the code, but didn't come across anything I considered to be a deal breaker in this context.

**Comment by aclum** [comment]
@turbomam please rereview, i believe this is ready to merge.

**Pull request merged**